b2h5py/blosc2.py

```diff
@@ -189,3 +190,45 @@ def opt_slice_read(dataset, slice_, new_dtype=None):
             "Selection is not suitable for Blosc2 optimized slicing")
 
     return opt_selection_read(dataset, selection, new_dtype)
+
+
+
+class B2Dataset:
+    """Allow to read data from a h5py dataset compressed with Blosc2 in an efficient way
+
+    Example:
+
+    .. code-block:: python
+
+        f = h5py.File("/path/to/file.h5", "r")
+        ds = B2Dataset(f["/hdf5/path/to/blocs2_compressed_dataset"])
+        data = ds[:10, :]
+        f.close()
+    """
+
+    def __init__(self, dataset: h5py.Dataset):
+        if not isinstance(dataset, h5py.Dataset):
+            raise ValueError("dataset must be a h5py.Dataset")
+        self.__dataset = dataset
+
+    @property
+    def dataset(self) -> h5py.Dataset:
+        """The h5py dataset this instance gives access to"""
+        return self.__dataset
+
+    @h5cached_property
+    def is_b2_fast_access(self) -> bool:
+        """Whether or not Blosc2 fast slicing access is enabled"""
+        return opt_slicing_dataset_ok(self.dataset)
+
+    def __getitem__(self, args):
```

Comment:
This can be much simplified if the `_blosc2_opt_slicing_ok` property is provided (equivalent to `is_b2_fast_access`), since the operation would be reduced to just attempting `opt_slice_read` (which does all the checking, also for `opt_slicing_enabled`, which is missing here) & catching `NoOptSlicingError` (within the `with` block), then calling the original getitem if it's caught. I.e. virtually what `b2h5py.patch.B2Dataset___getitem__` does. :wink:

Reply:
Indeed, best to avoid copies.